subsys/bluetooth/mesh/vnd/silvair_enocean_srv.c

```diff
@@ -111,14 +114,13 @@ static void state_machine_button_process(
 	switch (action) {
 	case BT_ENOCEAN_BUTTON_PRESS:
 		button->state = BT_MESH_SILVAIR_ENOCEAN_STATE_WAIT;
+		button->start_ts = k_uptime_get();
 		button->target = on_off;
 		k_work_reschedule(&button->timer,
 			K_MSEC(CONFIG_BT_MESH_SILVAIR_ENOCEAN_WAIT_TIME));
 		break;
 	case BT_ENOCEAN_BUTTON_RELEASE:
-		if (button->state != BT_MESH_SILVAIR_ENOCEAN_STATE_DELTA ||
-		    !k_work_delayable_is_pending(&button->timer) ||
-		    button->target != on_off) {
+		if (!button->onoff_disabled || button->target != on_off) {
```

Comment:
Could you explain why state check and timer check didn't work? If I see correctly, `onoff_disable` is true while state is `BT_MESH_SILVAIR_ENOCEAN_STATE_DELTA` and set to false when either:
- state is `BT_MESH_SILVAIR_ENOCEAN_STATE_DELTA` and timer won't be running anymore (same as it was before), or
- whenever button is released.

Reply:
I think I have a full description in the JIRA issue on this part. In short: After the initial sent delta message (ts 400ms) the delayed work is not reschedueled. It is first rescedueled on the last message call from the update CB. In the period between these two events some time (measured approx 1 sec) will elapse where the delayed work is not pending. If a button is released during this period it will cause a onoff message to be issued, which to my understanding is incorrect behavior.

Reply:
Could `!k_work_delayable_is_pending(&button->timer)` be replaced with `button->tick_count >= MAX_TICKS`?

Reply:
Could this be solved by just checking for STATE_DELTA if we go back to STATE_IDLE in the new `else` in the timeout? You'd probably need to remove the STATE_DELTA early exit in the update callback to make this work, but as far as I can tell, that is redundant, as the retransmit count is set in send_delta. The comment for it is outdated, at least

Reply:
I still don't see the purpose of onoff_disabled, it seems to be true if and only if state == DELTA, why do you need both? When would state != DELTA but onoff_disabled == true or vice versa? Am I missing something. You "If a button is released during this period" case would be with state == DELTA, no?

Reply:
Currently STATE_DELTA isn't reset until the button is released, but onoff_disabled is disabled when the timer stops repeating. This looks like it might also be a bug? We can't guarantee that we receive the "released" button event